android/guava/src/com/google/common/cache/LocalCache.java

```diff
@@ -1946,24 +1945,16 @@ static class Segment<K, V> extends ReentrantLock {
       this.statsCounter = checkNotNull(statsCounter);
       initTable(newEntryArray(initialCapacity));
 
-      keyReferenceQueue = map.usesKeyReferences() ? new ReferenceQueue<K>() : null;
+      keyReferenceQueue = map.usesKeyReferences() ? new ReferenceQueue<>() : null;
 
-      valueReferenceQueue = map.usesValueReferences() ? new ReferenceQueue<V>() : null;
+      valueReferenceQueue = map.usesValueReferences() ? new ReferenceQueue<>() : null;
 
       recencyQueue =
-          map.usesAccessQueue()
-              ? new ConcurrentLinkedQueue<ReferenceEntry<K, V>>()
-              : LocalCache.<ReferenceEntry<K, V>>discardingQueue();
+          map.usesAccessQueue() ? new ConcurrentLinkedQueue<>() : LocalCache.discardingQueue();
 
-      writeQueue =
-          map.usesWriteQueue()
-              ? new WriteQueue<K, V>()
-              : LocalCache.<ReferenceEntry<K, V>>discardingQueue();
+      writeQueue = map.usesWriteQueue() ? new WriteQueue<>() : LocalCache.discardingQueue();
 
-      accessQueue =
-          map.usesAccessQueue()
-              ? new AccessQueue<K, V>()
-              : LocalCache.<ReferenceEntry<K, V>>discardingQueue();
+      accessQueue = map.usesAccessQueue() ? new AccessQueue<>() : LocalCache.discardingQueue();
```

Comment:
**Performance Issue**: The simplification of the queue initialization expressions using diamond operators improves code readability and conciseness. This change aligns with modern Java practices, making the codebase easier to understand and maintain. The transition to using diamond operators where possible is a recommended practice in Java to reduce verbosity without sacrificing clarity or type safety. <br> **Fix**: Continue using diamond operators for similar initializations throughout the codebase to maintain consistency and improve readability. Review other parts of the code where explicit type arguments are used during initialization and assess if they can be safely replaced with diamond operators without losing clarity or type safety. <br> **Code Suggestion**: 
 ```
 +      accessQueue = map.usesAccessQueue() ? new ConcurrentLinkedQueue<>() : LocalCache.discardingQueue();
 ```